app/models/submission.rb

```diff
@@ -29,4 +31,8 @@ class Submission < ApplicationRecord
       transitions from: :started, to: :completed
     end
   end
+
+  def application_reference
+    request_body["applicationRef"]
```

Comment:
Is this a PP specific reference? The submission model should be neutral and any service specific stuff in a class that represents the submission using a consistent interface.

Reply:
Yes currently it is but we can extract / modify it when adding PlanX submissions